lib/rules/no-restricted-imports.js

```diff
@@ -248,26 +216,31 @@ module.exports = {
          */
         function checkNode(node) {
             const importSource = node.source.value.trim();
-            const importNames = node.specifiers ? node.specifiers.reduce((set, specifier) => {
+            const importNames = node.specifiers ? node.specifiers.reduce((map, specifier) => {
+                let name;
+                const specifierData = { loc: specifier.loc };
+
                 if (specifier.type === "ImportDefaultSpecifier") {
-                    set.add("default");
+                    name = "default";
                 } else if (specifier.type === "ImportNamespaceSpecifier") {
-                    set.add("*");
+                    name = "*";
                 } else if (specifier.imported) {
-                    set.add(specifier.imported.name);
+                    name = specifier.imported.name;
                 } else if (specifier.local) {
-                    set.add(specifier.local.name);
+                    name = specifier.local.name;
                 }
-                return set;
-            }, new Set()) : new Set();
 
-            if (isRestrictedForEverythingImported(importSource, importNames)) {
-                reportPathForEverythingImported(importSource, node);
-            }
+                if (map.has(name)) {
+                    map.set(name, map.get(name).concat(specifierData));
+                } else {
+                    map.set(name, [specifierData]);
+                }
```

Comment:
Maybe we should wrap this in `if (name) {}`.

Reply:
An alternative for the `concat` line is `map.get(name).push(specifierData);` without `set`, though this also looks fine to me!

Reply:
> Maybe we should wrap this in `if (name) {}`.

Yes, you're right. You even mentioned above that could be the case where specifier might not have `imported` or `local` property 👍

Reply:
> An alternative for the `concat` line is `map.get(name).push(specifierData);` without `set`, though this also looks fine to me!

Yep, `push` is much faster 👍